### PR TITLE
feat: add items in context

### PR DESCRIPTION
**Pull request by SamuelAlev**
- Added `isEditing`, `isAuthenticated`, `currentLanguage`, `defaultLanguage` and `projectId` to the context
- Added helpers to build links

### CODE REVIEW

File: packages/app-bridge/src/helpers/link.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    documentCategory?: DocumentCategory;
+    language: string;
+    portalToken: Nullable<string>;
+}) => {
+    const categoryUrlPart = documentCategory ? `${documentCategory.slug}/` : '';
+    if (portalToken) {
+        return `/d/${portalToken}/${language}/${document.slug}#/${categoryUrlPart}${documentPage.slug}`;
+    }
+
+    return `/document/${document.id}/${language}#/${categoryUrlPart}${documentPage.slug}`;
```
Comment (hochreutenerl) [inline packages/app-bridge/src/helpers/link.ts:55]: Please use `getUrlFragment` for the hash and following parts. Same for line 52 above

Comment (SamuelAlev) [reply packages/app-bridge/src/helpers/link.ts:55]: We are in Brand SDK repository, not Web App. Did you change some routing logic recently?

Comment (hochreutenerl) [reply packages/app-bridge/src/helpers/link.ts:55]: Yes. The `categoryUrlPart` should be `-/` if there is no `documentCategory`
From my current perspective we should avoid generating these urls externally.

File: packages/app-bridge/src/AppBridgeBlock.ts [hunk 2]
```diff
@@ -204,22 +204,10 @@ export interface AppBridgeBlock<
      */
     closeAssetChooser(): void;
 
-    /**
```
Comment (SamuelAlev) [inline packages/app-bridge/src/AppBridgeBlock.ts:207]: To be done later

File: packages/app-bridge/src/AppBridgeTheme.ts [hunk 3]
```diff
@@ -252,7 +259,7 @@ export interface AppBridgeTheme<
 
     updateBrandportalLink(brandportalLink: Partial<BrandportalLink>): Promise<Partial<BrandportalLink> | null>;
 
-    getCoverPage(): Promise<CoverPage>;
+    getCoverPage(language?: string): Promise<CoverPage>;
```
Comment (mike85) [inline packages/app-bridge/src/AppBridgeTheme.ts:262]: Does the caller of this function need to have control over the `language` parameter?

File: packages/app-bridge/src/react/useCoverPage.ts [hunk 4]
```diff
@@ -26,7 +26,7 @@ export const useCoverPage = (
 
     const fetchCoverPage = useCallback(async () => {
         setIsLoading(true);
-        setCoverPage(await appBridge.getCoverPage());
+        setCoverPage(await appBridge.getCoverPage(appBridge.context('currentLanguage').get()));
```
Comment (mike85) [inline packages/app-bridge/src/react/useCoverPage.ts:29]: Couldn’t we just call `getCoverPage()` and use the updated context value from within the appBridge function?

Comment (SamuelAlev) [reply packages/app-bridge/src/react/useCoverPage.ts:29]: By default it uses the context in the implementation, so it is not necessary here.

Screenshot 2023-11-29 at 14 36 41

Will remove it 👍

File: packages/app-bridge/src/AppBridgeTheme.ts (hunk 3 above)
Comment (SamuelAlev) [reply packages/app-bridge/src/AppBridgeTheme.ts]: Removed

**Pull request merged**